crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -39,9 +39,13 @@ impl FormatNodeRule<MatchCase> for FormatMatchCase {
                         write!(f, [text("case"), space()])?;
 
                         if is_match_case_pattern_parenthesized(item, pattern, f.context())? {
```

Comment:
Can you explain why calling `pattern.format` isn't sufficient? Why can the pattern and the `MatchCase` both have trailing opening parentheses comments.

Reply:
Consider:

```python
match thing:
    case ( # outer
        [  # inner
            1
        ]
    )
```

The `# outer` is attached to the `MatchCase`, but the `# inner` is part of the pattern itself. (Not all patterns have this behavior.)

Reply:
Hmm, so it depends on whether the pattern has its own parentheses? How does this work if you have nested pattern, which doesn't go through the match case formatting?

```
match thing:
	case [
		( # outer
			[ # inner
				1, 
			]
		)
	]: ...

Reply:
Both of these cases work as expected as of the following branch (#6801):

```python
match thing:
    case [
		( # outer
			[ # inner
				1,
			]
		)
	]: ...
    case [ # outer
		( # inner outer
			[ # inner
				1,
			]
		)
	]: ...
```

`# inner outer` gets assigned as a leading end-of-line comment via our standard parenthesized comment handling, which then gets rendered as an open parenthesis comment.

Reply:
Actually, let me confirm that that's _why_ it works, hang on.

Reply:
Err, I think it works because it's a leading end-of-line comment which we always treat as an open parenthesis comment. (And the comments on the inner square brackets work because they're correctly marked as dangling for the sequence pattern type.)

Reply:
I guess we kind of get this wrong:

```python
    case [ # outer
        # own line
		( # inner outer
			[ # inner
				1,
			]
		)
	]:
        pass
```

It renders as:

```python
    case [  # outer
        (
            # own line
            # inner outer
            [  # inner
                1,
            ]
        )
    ]:
```

Reply:
We might be able to remove this, I'm not sure yet, I'll explore it.

Reply:
(I'd thought we wanted to preserve the parentheses here always which complicated this a bit.)

Reply:
Ok, I was able to remove the dangling comment from `match_case`.